### PR TITLE
ORM page fixes....

### PR DESCRIPTION
**Pull request by leonardteo**
Updated two pages:
- crud.html
- models.html
# crud.html

Find first/last - the example given was actually incorrect. Based on what I see in 1.01 core, a find('first'/'last..) will always return you a record by sorting by primary key. Adding an orderby clause doesn't seem to do anything.

There were errors with some of the code when passing in where clauses...
$entries = Model_Article::find('all', array(
    'where' => array(
        array('category_id', 1),    //This is incorrect and will throw an exception
    ),
    'order_by' => array('date' => 'desc'),
));

$entries = Model_Article::find('all', array(
    'where' => array(
        array('category_id' => 1),    //This is correct
    ),
    'order_by' => array('date' => 'desc'),
));

Also....I changed the $entry to $entries (plural) for methods that return an array.

Added some descriptions 
# models.html

Fixed where clause

### CODE REVIEW

File: general/models.html [hunk 1]
```diff
@@ -69,13 +69,12 @@
...
-$entry = Model_Article::find('all');
+		<pre class="php"><code>// find all articles - returns an array of Model_Article
+$entries = Model_Article::find('all');
 
-// find all articles from category 1 order descending by date
+// find a Model_Article from category 1 with the greatest primary key (id)
 $entry = Model_Article::find('last', array(
-    'where' => array('category_id', 1),
-    'order_by' => array('date', 'desc')
+    'where' => array('category_id' =&gt; 1)
```
Comment (frankdejonge) [inline general/models.html:96]: This doesn't seem right... shouldn't it be 'where' => array('category_id', '=&gt;', 1)

File: packages/orm/crud.html [hunk 2]
```diff
@@ -83,24 +83,27 @@ $entry = Model_Article::find(array(2, 5));</code></pre>
...
+				<pre class="php"><code>// find all articles - returns array of Model_Article
+$entries = Model_Article::find('all');
 
-// find all articles from category 1 order descending by date
-$entry = Model_Article::find('all', array(
+// find all articles from category 1 order descending by date - returns array of Model_Article
+$entries = Model_Article::find('all', array(
 	'where' =&gt; array(
-		array('category_id', 1),
+		array('category_id' =&gt; 1),
```
Comment (frankdejonge) [inline packages/orm/crud.html:122]: same here

File: general/models.html (hunk 1 above)
Comment (leonardteo) [inline general/models.html:96]: Yup.  '&amp;gt;' (HTML greater than symbol) will render as > on a browser so the above line will render:
'where' => array('category_id' => 1)

Do you want all PHP code in raw PHP and not HTML'ized?

Comment (frankdejonge) [inline general/models.html:96]: Sorry misread...

**Comment by leonardteo** [comment]
I added a real world example to the Tasks page.

**Comment by leonardteo** [comment]
Fixed another error. Customer validation method needs to be declared as static or the app fails.
Added more clarity around setting custom error messages in the language file.

**Comment by WanWizard** [comment]
This can not be merged anymore. It this still valid, or can it be closed, @jschreuder?

**Comment by leonardteo** [comment]
This was so long ago the changes are irrelevant now. Not sure why they
weren't merged back then.

L.

2011/12/9 Harro Verton <
reply@reply.github.com

**Pull request closed without merging**